### PR TITLE
amqp kit stop method

### PR DESCRIPTION
**Pull request by akolomietc**

### CODE REVIEW

File: amqp-kit/api.go [hunk 1]
```diff
@@ -81,6 +84,10 @@ func (s *Server) Serve() error {
 	return nil
 }
 
+func (s *Server) Stop(noWait bool) error {
```
Comment (avdva) [inline amqp-kit/api.go]: why do you need unused parameter?

Comment (akolomietc) [reply amqp-kit/api.go]: sorry, mistake )

File: amqp-kit/api.go [hunk 2]
```diff
@@ -27,29 +27,34 @@ type Config struct {
...
 
 type Server struct {
-	Conn *amqp.Connection
-	qk   map[string]map[string]func(deliv *amqp.Delivery)
-	subs []SubscribeInfo
+	Conn    *amqp.Connection
+	qk      map[string]map[string]func(deliv *amqp.Delivery)
+	subs    []SubscribeInfo
+	ch      *amqp.Channel
+	closeCh chan *amqp.Error
```
Comment (avdva) [inline amqp-kit/api.go]: is it used?

File: amqp-kit/api.go [hunk 3]
```diff
@@ -59,20 +64,23 @@ func (s *Server) Serve() error {
...
 		}
 
 		go func(q string) {
-			for d := range msgs {
+			for {
+				d, ok := <-msgs
+				if !ok {
+					break
+				}
 				if _, exist := s.qk[q][d.RoutingKey]; !exist {
```
Comment (avdva) [inline amqp-kit/api.go]: if handler, exist := s.qk[q][d.RoutingKey]; !exist {
...
} else {
handler(&d)
}
?

File: amqp-kit/api.go (hunk 2 above)
Comment (akolomietc) [reply amqp-kit/api.go]: no (

File: amqp-kit/api.go [hunk 4]
```diff
@@ -59,28 +62,36 @@ func (s *Server) Serve() error {
...
+				d, ok := <-msgs
+				if !ok {
+					break
+				}
+
+				if handler, exist := s.qk[q][d.RoutingKey]; !exist {
 					d.Nack(false, false)
 					log.Printf(`subscribe key not found %s`, d.RoutingKey)
-
 					continue
```
Comment (avdva) [inline amqp-kit/api.go]: no need for 'continue' anymore

**Review by avdva — APPROVED** [review]

**Pull request merged**